### DISCUSSION TITLE
Custom Retronas Directories/Symlinks

### DISCUSSION THREAD

**Discussion by zacaronii** [retronas/discussions#130] [Q&A]
I really appreciate all of the options retronas offers, but I wondered if I could setup my own custom directories with symlinks to the generic roms folders. I know I CAN, but I wanted to make sure nothing in the ansible playbooks would cause any custom setups to get overwritten in the future.

For instance, I know you have an emudeck setup, but I have been using retrodeck and would like to configure a directory for that. In this particular case, I think what I'd actually like to do is just use the emudeck setup as a base, rename the primary directory to "retrodeck" and make some tweaks to some of the folders from there to customize it for retrodeck.

Actually, to expand on this question a bit, if I make customizations to any of the directories is there a possibility for that to get overwritten if I'm not actually going in and rerunning the ansible playbook for that specific share?

**Comment by sairuk** [comment]
You can run a custom playbook from the Update menu if you wanted to build one, however when it comes to the software dirs and specifically the `retronas_systems.yml` file (which ideally you'd want to edit) there is every possibility future changes could impact any custom setup but so long as your playbook is completely self-contained it should be OK.

i've added support for a retrodeck folder in the testing branch if you want to try it out

**Reply by sairuk** [reply]
i've also added support to my update checking tool, which checks against the [es_systems.xml](https://raw.githubusercontent.com/RetroDECK/ES-DE/refs/heads/main/resources/systems/unix/es_systems.xml) for supported systems, so hopefully that is correct for RD

**Reply by zacaronii** [reply]
That's a really neat tool! I'll hop on the testing branch and try the retrodeck setup here in a little bit. 

I love this project by the way. Thank you for all of the work you've put in to it. Do you have any preference on how people who want to support the project contribute to it?

**Reply by sairuk** [reply]
> Do you have any preference on how people who want to support the project contribute to it?

not really atm, i suppose if anything code, tickets, tutorials but ultimately just enjoy it

**Comment by zacaronii** [comment]
I installed the retrodeck folder from the testing branch, and I did a comparison of the default directories from retrodeck and retronas. I provided my csv with my comparison, so you can check my work if you'd like. However, these are the changes I found necessary to align the directory structure more accurately with the default from retrodeck. This is specifically for the roms folder. I'll try to actually learn how to use git, so I can be more useful and provide an actual pull request in the future.

Suggestion
--
REMOVE ags ->   ../../roms/commodore/amiga
REMOVE android ->   ../../roms/platform/android
REMOVE androidapps ->   ../../roms/platform/android/apps
REMOVE androidgames ->   ../../roms/platform/android/games
REMOVE atarijaguarcd ->   ../../roms/atari/jaguar/cd
REMOVE coco ->   ../../roms/tandy/trs80colorcomputer1
REMOVE dragon32 -> ../../roms/dragon
REMOVE flash -> ../../roms/other/flash
REMOVE fpinball ->   ../../roms/pinball/fpinball 
REMOVE mame-advmame ->   ../../roms/mame/mame
ADD mastersystem ->   ../../roms/sega/mastersystem
REMOVE megadrivejp ->   ../../roms/sega/megadrive
REMOVE model3 ->   ../../roms/sega/model3
REMOVE ngage ->   ../../roms/nokia/n-gage
REMOVE oric ->   ../../roms/oric/tangerine
ADD pc -> ../../roms/ibm/pcxt
REMOVE ps4 ->   ../../roms/sony/playstation4
REMOVE samcoupe ->   ../../roms/milesgordon/samcoupe
REMOVE sneshd ->   ../../roms/nintendo/superfamicom
ADD stv -> ../../roms/sega/stv
REMOVE symbian ->   ../../roms/symbian/symbianos
REMOVE tanodragon -> ../../roms/dragon
REMOVE triforce ->   ../../roms/sega/triforce
REMOVE trs-80 ->   ../../roms/tandy/trs80
REMOVE vpinball ->   ../../roms/pinball/vpinball
REMOVE xbox360 ->   ../../roms/microsoft/xbox360
REMOVE zmachine ->   ../../roms/infocom/zmachine
REMOVE zxnext ->   ../../roms/sinclair/zxspectrum
 

[retrodeck_roms_directories.csv](https://github.com/user-attachments/files/18054311/retrodeck_roms_directories.csv)

**Reply by sairuk** [reply]
Thanks for taking a look.

All the items (that i've checked) that you've marked for removal are valid directories in their es systems configuration so they wont be removed from our end. We need a base reference and the es systems files is our best/current option. If they are not actually creating the directories in the project release there may be a reason for it but we're not going to try and manage that from here.

pc I have now linked to pcxt as it should've been, oversight on my part, i meant to loopback around to that and confirm but seems I forgot to get back to it.

mastersystem and stv links should have been created as part of the install, if they weren't we probably want to understand why. Although perhaps i've misinterpreted what your data is saying.

my resulting installation had this created automatically
```
lrwxrwxrwx 1 root root 28 Dec  8 18:39 mastersystem -> ../../roms/sega/mastersystem
lrwxrwxrwx 1 root     root       19 Dec  8 18:39 stv -> ../../roms/sega/stv
```

**Reply by zacaronii** [reply]
Awesome! Thank you for your thorough response. I'll double check things on my end when I get a moment and see if I can figure out whether I just made a mistake or if the links actually didn't get created for those other two. 

edit: Just to confirm, it seems like the two that didn't get created were my fault. I was playing around with a retrodeck directory in my retronas before you released the official setup in the testing branch. I thought I cleaned up all of the old directories, but I must have left the mastersystem and stv directories that I had created. Once I removed them and ran the ansible playbook again they were linked correctly.

**Reply by zacaronii** [reply]
Also, I'll try to inquire with the retrodeck team on the es systems config, once I double check my retrodeck setup, to see if there's an reason for those entries that aren't actually in the directory structure or if they could be removed.

**Reply by sairuk** [reply]
cool, if they mention there is a different reference we should look at, we can revisit

**Reply by XargonWan** [reply]
> Thanks for taking a look.
> 
> All the items (that i've checked) that you've marked for removal are valid directories in their es systems configuration so they wont be removed from our end. We need a base reference and the es systems files is our best/current option. If they are not actually creating the directories in the project release there may be a reason for it but we're not going to try and manage that from here.
> 
> pc I have now linked to pcxt as it should've been, oversight on my part, i meant to loopback around to that and confirm but seems I forgot to get back to it.
> 
> mastersystem and stv links should have been created as part of the install, if they weren't we probably want to understand why. Although perhaps i've misinterpreted what your data is saying.
> 
> my resulting installation had this created automatically
> 
> ```
> lrwxrwxrwx 1 root root 28 Dec  8 18:39 mastersystem -> ../../roms/sega/mastersystem
> lrwxrwxrwx 1 root     root       19 Dec  8 18:39 stv -> ../../roms/sega/stv
> ```

Greetings, I'm Xargon, RetroDECK founder.
What you suggest is correct: I wouldn't remove any folder, or, I would make them dynamic as if RetroDECK is not supporting a specific system now, it doesn't mean that it will not in the future, as we wish to have at least one emulator for each system as a goal.

So if you hardcode them, better to leave them in place, if you parse them from `es_systems.xml` just count the non commented systems and you will be good if the file is grabbed dynamically.
We update it reflecting the state of our supported systems, for example in cooker (our development branch) we just added the `portmaster` system.

This file is of course coming from the official ES-DE, but at this point ours is customised with our own needs.

Thanks a lot you all for supporting RetroDECK, I greatly appreciate that. And feel free to reach us out :)

**Reply by zacaronii** [reply]
Hey, Xargon! Thank you for the quick follow up on this topic. Also, thank you for all of the work you are doing on RetroDECK.

If I understand correctly, it looks like almost all of the directories I mentioned as needing to be removed are actually commented out in the following file under the linux directory. [es_systems.xml](https://raw.githubusercontent.com/RetroDECK/ES-DE/refs/heads/main/resources/systems/linux/es_systems.xml)

The only ones I see that are not commented out in this .xml, that don't seem to be generated in the default RetroDECK roms directory, are the 3 android entries. Would it be acceptable to comment these entries out?

> ~~REMOVE ags -> ../../roms/commodore/amiga~~
> REMOVE android -> ../../roms/platform/android
> REMOVE androidapps -> ../../roms/platform/android/apps
> REMOVE androidgames -> ../../roms/platform/android/games
> ~~REMOVE atarijaguarcd -> ../../roms/atari/jaguar/cd~~
> ~~REMOVE coco -> ../../roms/tandy/trs80colorcomputer1~~
> ~~REMOVE dragon32 -> ../../roms/dragon~~
> ~~REMOVE flash -> ../../roms/other/flash~~
> ~~REMOVE fpinball -> ../../roms/pinball/fpinball~~ 
> ~~REMOVE mame-advmame -> ../../roms/mame/mame~~
> ~~ADD mastersystem -> ../../roms/sega/mastersystem~~
> ~~REMOVE megadrivejp -> ../../roms/sega/megadrive~~
> ~~REMOVE model3 -> ../../roms/sega/model3~~
> ~~REMOVE ngage -> ../../roms/nokia/n-gage~~
> ~~REMOVE oric -> ../../roms/oric/tangerine~~
> ~~ADD pc -> ../../roms/ibm/pcxt~~
> ~~REMOVE ps4 -> ../../roms/sony/playstation4~~
> ~~REMOVE samcoupe -> ../../roms/milesgordon/samcoupe~~
> REMOVE sneshd -> ../../roms/nintendo/superfamicom (I don't see this in the .xml at all, unless I'm overlooking something) 
> ~~ADD stv -> ../../roms/sega/stv~~
> ~~REMOVE symbian -> ../../roms/symbian/symbianos~~
> ~~REMOVE tanodragon -> ../../roms/dragon~~
> ~~REMOVE triforce -> ../../roms/sega/triforce~~
> ~~REMOVE trs-80 -> ../../roms/tandy/trs80~~
> ~~REMOVE vpinball -> ../../roms/pinball/vpinball~~
> ~~REMOVE xbox360 -> ../../roms/microsoft/xbox360~~
> ~~REMOVE zmachine -> ../../roms/infocom/zmachine~~
> ~~REMOVE zxnext -> ../../roms/sinclair/zxspectrum~~

**Reply by sairuk** [reply]
Yeh there are ~6 `es_systems.xml` files in the retrodeck repo, @zacaronii your thinking is we should be using the linux one that has stuff commented out. 

Although assuming the other `es_systems.xml` entries are valid targets we'd need to cover all scenarios since we're a storage endpoint in this case, so having all directories available makes sense for our end.

**Reply by zacaronii** [reply]
Yeah, that makes perfect sense. I don't know for sure if those other entries are utilized by the project at the moment, or if they are just entries from the official ES-DE repo. The docs currently state that the RetroDECK project only supports Steam OS and limited Linux desktop support, at the moment. 

If RetroDECK is only leveraging the `es_systems.xml` in the linux directory, then perhaps it could make sense to use that one. 

However, if there are no adjustments that can be made that make sense for both projects, then we will just have to leave it at that. 👍 Honestly, there are several ways I could see to work around this on the user side, if that ends up being necessary.

**Reply by zacaronii** [reply]
Hey @XargonWan , sorry to bump you on this again. I appreciate you taking the time to respond to the inquiries here. I just wanted to follow up again to see if you could confirm our understanding of how RetroDECK is using the .xml files, so we can assure that Retronas is grabbing the most accurate information. 

To be clear, Retronas is parsing this file, right now.  [RetroDECK/ES-DE/refs/heads/main/resources/systems/unix/es_systems.xml](https://raw.githubusercontent.com/RetroDECK/ES-DE/refs/heads/main/resources/systems/unix/es_systems.xml)

However, this one appears to better match your description of having certain systems commented out. [RetroDECK/ES-DE/refs/heads/main/resources/systems/linux/es_systems.xml](https://raw.githubusercontent.com/RetroDECK/ES-DE/refs/heads/main/resources/systems/linux/es_systems.xml)

Is it an accurate understanding that RetroDECK is using the one in the linux directory to represent the supported systems, or are the other .xml files maintained, as well?

**Reply by sairuk** [reply]
just so we are all on the same page

we were checking only the unix systems originally as it seemed complete. we now check all the entries of all es_systems.xml files for the systems in the resources directory using our [check_systems](https://github.com/retronas/check-systems) tool to compare against other projects we already support.

we take those results and add them to our mapping file [retronas_systems](https://github.com/retronas/retronas/blob/testing/ansible/retronas_systems.yml) because we have a directory structure that is expected to provide the same files across multiple projects using symlinks.

i'd assume the linux file is the correct one, now that its clear the commented systems are not used by the project intentionally. we can pull back those entires in our mapping file easily enough if so.
